project/recursive_sorting.py

```diff
@@ -44,8 +49,42 @@ def merge_sort_in_place(arr, l, r):
 
 # TO-DO: implement the Quick Sort function below USING RECURSION
 def quick_sort( arr, low, high ):
-
+    if low < high:
+        split = partition(arr, low, high)
+        quick_sort( arr, low, split - 1)
+        quick_sort( arr, split + 1, high)
     return arr
+    
+def partition(arr, low, high):
+
+    pivot = arr[low]
+    leftmark = low + 1
+    rightmark = high
+    done = False
+
+    while not done:
+        while leftmark <= rightmark and arr[leftmark] <= pivot:
+            leftmark = leftmark + 1
+        while arr[rightmark] >= pivot and rightmark >= leftmark:
+            rightmark = rightmark - 1
+        
+        if rightmark < leftmark:
+            done = True
+        else:
+            temp = arr[leftmark]
```

Comment:
this also works for swaps

arr[leftmark],  arr[rightmark]= arr[rightmark], arr[leftmark]